server/conn.go

```diff
@@ -521,6 +522,8 @@ func (cc *clientConn) readOptionalSSLRequestAndHandshakeResponse(ctx context.Con
 				return err
 			}
 		}
+	} else if config.GetGlobalConfig().Security.RequireSecureTransport {
+		return errors.New("secure transport is required")
```

Comment:
I think we should add the same error that is used in MySQL: https://dev.mysql.com/doc/refman/8.0/en/server-error-reference.html#error_er_secure_transport_required